roles/rke2_server/tasks/first_server.yml

```diff
@@ -1,14 +1,48 @@
 ---
 
-- name: Install
-  include_tasks: install.yml
+- name: Add manifest files
+  include_role:
+    name: rke2_common
+    tasks_from: add-manifest-addons.yml
+  when: manifest_config_file_path is defined
 
 - name: Start rke2-server
   ansible.builtin.systemd:
     name: rke2-server
     state: started
     enabled: yes
 
+- name: Wait for k8s apiserver
+  wait_for:
+    host: "{{ kubernetes_api_server_host }}"
+    port: "6443"
+    state: present
+    timeout: 300
+
+- name: Wait for kubelet process to be present on host
+  ansible.builtin.shell: >-
+    ps -C kubelet -F -ww --no-headers
+  register: kubelet_check
+  until: kubelet_check.rc == 0
+  retries: 20
+  delay: 10
+  changed_when: false
+
+- name: Extract the hostname-override parameter from the kubelet process
+  set_fact:
+    kubelet_hostname_override_parameter: "{{ kubelet_check.stdout | regex_search('\\s--hostname-override=((([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\\-]*[A-Za-z0-9]))\\s' , '\\1') }}"
+
+- name: Wait for node to show Ready status
```

Comment:
@dgvigil This _should_ fix the inventory_hostname thing. Not the best approach but this seems to be fine with both cloud_provider and manual node names